lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -182,4 +188,59 @@
 
         return -1;
     });
+
+    platform.registerFunction(FunctionsEnum.ArrayFilter, function (callbackfn, thisArg) {
+        // ECMAScript 2017 #sec-array.prototype.filter
+        "use strict";
+
+        if (this === null || this === undefined) {
+            __chakraLibrary.raiseThis_NullOrUndefined("Array.prototype.filter");
+        }
+
+        let o;
+        let len
+        if (__chakraLibrary.isArray(this)) {
```

Comment:
why not do Array.isArray()?

Reply:
This calls the same function, but from a safe location so that we are unaffected if the user replaces Array.isArray

Reply:
ah - thanks,

---
In reply to: [187739035](https://github.com/Microsoft/ChakraCore/pull/5137#discussion_r187739035) [](ancestors = 187739035)